utils/AsyncIterableTask.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable functional/prefer-readonly-type */
+/* eslint-disable @typescript-eslint/array-type */
+/* eslint-disable functional/immutable-data */
+/* eslint-disable functional/no-let */
+/* eslint-disable prefer-const */
+// -------------------------------------------------------------------------------------
+// model
+// -------------------------------------------------------------------------------------
+
+import * as T from "fp-ts/lib/Task";
+import * as TE from "fp-ts/lib/TaskEither";
+
+import { mapAsyncIterable } from "@pagopa/io-functions-commons/dist/src/utils/async";
+import { pipe } from "fp-ts/lib/function";
+
+/**
+ * @category model
+ * @since 2.0.0
+ */
+export type AsyncIterableTask<A> = T.Task<AsyncIterable<A>>;
+
+export const fromAsyncIterable = <A>(
+  a: AsyncIterable<A>
+): AsyncIterableTask<A> => T.of(a);
+
+/**
+ * `map` can be used to turn functions `(a: A) => B` into functions `(fa: F<A>) => F<B>` whose argument and return types
+ * use the type constructor `F` to represent some computational context.
+ *
+ * @category Functor
+ * @since 2.0.0
+ */
+export const map: <A, B>(
+  f: (a: A) => B
+) => // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+(fa: AsyncIterableTask<A>) => AsyncIterableTask<B> = f => fa =>
+  pipe(
+    fa,
+    T.map(_ => mapAsyncIterable(_, f))
+  );
+
+/**
+ * Process an AsyncIterableTask and return an array of results
+ */
+export const fold = <A>(fa: AsyncIterableTask<A>): T.Task<ReadonlyArray<A>> =>
+  pipe(
+    fa,
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    T.chain(_ => async () => foldIterableArray<A>(_)())
```

Comment:
In this case `async` is redundant.  
```suggestion
    T.chain(_ => () => foldIterableArray<A>(_)())
```

Reply:
Removed in [280f52a](https://github.com/pagopa/io-functions-services-cache/pull/32/commits/280f52a5d30b7dd67e5ac9556e1d271d9989e261)